bake/hclparser/hclparser.go

```diff
@@ -483,6 +500,13 @@ func Parse(b hcl.Body, opt Opt, val interface{}) hcl.Diagnostics {
 	return nil
 }
 
+func ValidateLabel(lbl string) error {
```

Comment:
This should not be public

Reply:
I checked on compose-spec and compose-go repos and have not seen any kind of validators for a service name. It's public so I can use it for compose in `bake` pkg. Do you prefer a dedicated pkg for this?

Reply:
Ah, that is for compose. It's a bit weird as `hclparser` package is supposed to be generic and not even know about the current schema. But I guess you want to detect it here so that you can send the `hcl.Diagnostics` error. Maybe the validator should be passed with parse opts?

Reply:
> But I guess you want to detect it here so that you can send the `hcl.Diagnostics` error.

Yes that's it

> Maybe the validator should be passed with parse opts?

Yeah sounds better like this. Let me know if changes lgty.